### PR TITLE
integrate clerk authentication in web

### LINKED ISSUE #1: web-login
As a user...from the car service center employee
I want to...identify myself in the webpage
So that...I can use other features of the webpage

Image

further improvement for next week:
creating a working authentication process to check the user and password. If a user doesn't have any account yet, then they can click on the register me part and will be offered with a new user registration page with creating the user profile along with the user name and password creation.

### PR DESCRIPTION
**Pull request by Arefin2023**

### CODE REVIEW

File: web/src/main.jsx [hunk 1]
```diff
@@ -11,26 +11,119 @@ import { AppointmentPage } from "./pages/AppointmentPage.jsx";
...
+          </>
+        ),
+      },
+      {
+        path: "/wip",
+        element: (
+          <>
+            <SignedIn>
+              <WipPage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
```
Comment (actyralf) [inline web/src/main.jsx]: You shouldn't use the UserButton in your routes. If you want to see it on all pages, embed it somewhere in your `RootLayout.jsx`

File: web/src/main.jsx [hunk 2]
```diff
@@ -11,26 +11,119 @@ import { AppointmentPage } from "./pages/AppointmentPage.jsx";
...
+          </>
+        ),
+      },
+      {
+        path: "/message",
+        element: (
+          <>
+            <SignedIn>
+              <MessagePage />
+              <UserButton afterSignOutUrl="http://localhost:5173/" />
```
Comment (actyralf) [inline web/src/main.jsx]: You will have to make this URL dynamic. When you deploy your code to `xyz.com`, `localhost:5173` will not be available. As this affects everyone, I will prepare a solution for that which I can share with the whole team.

**Review by actyralf — CHANGES_REQUESTED** [review]
Well done!

Please fix the `UserButton` issue from the comment below. You can ignore the issue with `localhost` for now, I will prepare a solution for everyone. You can then implement it later.

File: web/src/main.jsx (hunk 1 above)
Comment (Arefin2023) [reply web/src/main.jsx]: I've moved the userButton in RootLayout.jsx and commited it as "changes done". Should I merge the pull request now?

**Review by actyralf — APPROVED** [review]

**Pull request merged**